src/components/Datepicker/Datepicker.tsx

```diff
@@ -211,6 +215,12 @@ export const Datepicker: FC<DatepickerProps> = ({
     };
   }, [inputRef, datepickerRef, setIsOpen]);
 
+  useEffect(() => {
```

Comment:
now that I'm thinking about it, this effect is strictly used to sync the internal state with incoming data, BUT, 
1. it would only make sense to sync it if the date actually changed (check if we have to implement a diffing mechanism, eg: `if (dateChanged) {...}`, this also saves us from redundant internal renders (or infinite) 
2. receiving `dateValue` from props means we want this component to be `controlled`, which also means if the user clears this value, the internal state(s) should follow. 

Problem is for `2.` that we have a check: `if (dateValue)` that invalidates this

Reply:
@rluders @tulup-conner thoughts again?

Reply:
@SutuSebastian  That makes sense, we are speaking of a semi-controlled guy, although you didn't ask I'd love to speak my mind and keep moving forward adding even more to my contribution:

The approach will work, we would need to add on the docs that if he wants to clear with dateValue it would need to be null. 

`
useEffect(() => {
  if (dateValue !== undefined && dateValue !== selectedDate) {
    setSelectedDate(dateValue);
  }
}, [dateValue, selectedDate, setSelectedDate]);
`

Does that make sense?

Reply:
Okay, I added the diff mechanism and some storybook test cases for that @SutuSebastian. Thanks.